### PR TITLE
[DPE-5108] All dashboards in sharded cluster on one screen

### PR DESCRIPTION
**Pull request by Gu1nness**
## Issue

In sharded cluster, it's not possible to check all replicasets at once

## Solution

* Fix dashboard code
* Fix cluster value

### CODE REVIEW

**Comment by Gu1nness** [comment]
Screenshot 2024-08-16 at 10-02-44 MongoDB ReplSet Summary - 4 - Dashboards - Grafana
Screenshot 2024-08-16 at 10-02-56 MongoDB ReplSet Summary - 4 - Dashboards - Grafana
Screenshot 2024-08-16 at 10-03-24 MongoDB ReplSet Summary - 4 - Dashboards - Grafana
Screenshot 2024-08-16 at 10-03-32 MongoDB ReplSet Summary - 4 - Dashboards - Grafana
Screenshot 2024-08-16 at 10-03-40 MongoDB ReplSet Summary - 4 - Dashboards - Grafana

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 1]
```diff
@@ -468,7 +468,7 @@
             },
             "gridPos": {
                 "h": 8,
-                "w": 24,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: needed?

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 2]
```diff
@@ -931,7 +931,7 @@
...
+                "w": 8,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: needed?

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 3]
```diff
@@ -1103,7 +1103,7 @@
             },
             "gridPos": {
                 "h": 8,
-                "w": 12,
+                "w": 8,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: needed?

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 4]
```diff
@@ -1227,19 +1229,18 @@
             },
             "gridPos": {
                 "h": 8,
-                "w": 12,
-                "x": 12,
-                "y": 40
+                "w": 8,
+                "x": 0,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: same here

**Review by MiaAltieri — DISMISSED** [review]
Looks great, really looking forward to the demo 🤩

Some questions on the formatting, but from your screenshots the formatting looks great. So its fine not to revert the changes, only curious if those changes were purposeful or not

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 1 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Might not be needed indeed.
I will spin everything up again so I can check

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 2 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Might not be needed indeed.
I will spin everything up again so I can check

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 3 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Might not be needed indeed.
I will spin everything up again so I can check

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 4 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Might not be needed indeed.
I will spin everything up again so I can check

**Comment by Gu1nness** [comment]
@MiaAltieri Indeed those changes were unnecessary, I fixed it :)

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 5]
```diff
@@ -1226,20 +1228,19 @@
                 "rightLogBase": 1
             },
             "gridPos": {
-                "h": 8,
+                "h": 12,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: what about these ones

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 6]
```diff
@@ -2923,7 +2926,7 @@
                     "uid": "${prometheusds}"
                 },
                 "definition": "label_values({__name__=~\"pg_up|mysql_up|mongodb_up|proxysql_mysql_status_active_transactions\"}, replication_set)",
-                "hide": 2,
+                "hide": 0,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: is this needed

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 7]
```diff
@@ -2590,7 +2593,7 @@
                     "uid": "${prometheusds}"
                 },
                 "definition": "label_values(up{juju_model=\"$juju_model\",juju_model_uuid=\"$juju_model_uuid\",juju_application=\"$juju_application\"},juju_unit)",
-                "hide": 0,
+                "hide": 2,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: is this needed

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 8]
```diff
@@ -2620,7 +2623,7 @@
                     "uid": "${prometheusds}"
                 },
                 "definition": "label_values(up{juju_model=\"$juju_model\",juju_model_uuid=\"$juju_model_uuid\"},juju_application)",
-                "hide": 0,
+                "hide": 2,
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: is this needed

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 3 above)
Comment (MiaAltieri) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: just checking here

**Review by MiaAltieri — COMMENTED** [review]
some last little things I noticed, like before they are non blocking since the screenshot looks accurate

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 8 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Not needed but useful: It hides the `juju_application` selector which is not used anymore in this dashboard

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 7 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Not needed but useful: It hides the `juju_unit` selector which is not used anymore in this dashboard

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 6 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Yes, it allows to select the replicat_set value

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 5 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: fixed

**Review by MiaAltieri — DISMISSED** [review]

**Review by MiaAltieri — DISMISSED** [review]

File: src/charm.py [hunk 9]
```diff
@@ -175,7 +175,10 @@ def _mongo_scrape_config(self) -> List[Dict]:
                         "targets": [
                             f"{self.unit_host(self.unit)}:{Config.Monitoring.MONGODB_EXPORTER_PORT}"
                         ],
-                        "labels": {"cluster": self.app.name, "replication_set": self.app.name},
+                        "labels": {
+                            "cluster": self.get_config_server_name() or self.app.name,
+                            "replication_set": self.app.name,
```
Comment (Mehdi-Bendriss) [inline src/charm.py]: As discussed - let's proceed step by step on this one and only put it back when we confirm juju app / unit labels work as expected.

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 10]
```diff
@@ -2923,7 +2926,7 @@
                     "uid": "${prometheusds}"
                 },
                 "definition": "label_values({__name__=~\"pg_up|mysql_up|mongodb_up|proxysql_mysql_status_active_transactions\"}, replication_set)",
```
Comment (Mehdi-Bendriss) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json]: Let's please clean the dashboard from references to postgres/mysql etc..

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 11]
```diff
@@ -2580,126 +2584,6 @@
                 "skipUrlSync": false,
                 "type": "datasource"
             },
-            {
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json:2583]: QQ why was this removed

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 12]
```diff
@@ -2794,35 +2678,6 @@
                 "type": "query",
                 "useTags": false
             },
-            {
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json:2797]: same here

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json [hunk 13]
```diff
@@ -2922,73 +2777,15 @@
                 "datasource": {
                     "uid": "${prometheusds}"
                 },
-                "definition": "label_values({__name__=~\"pg_up|mysql_up|mongodb_up|proxysql_mysql_status_active_transactions\"}, replication_set)",
```
Comment (MiaAltieri) [inline src/grafana_dashboards/MongoDB_ReplSet_Summary.json:2925]: same here

**Review by MiaAltieri — COMMENTED** [review]
Looks like there are still 20 occurances of `service_name` in the json. Should those still be there?

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 11 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json:2583]: This is automatically generated by cos suite (+ here it had the wrong definition which made the dashboard fail in multiple situations.

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 12 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json:2797]: We're not using this filter anymore as it's a duplicate of `juju_application`.

File: src/grafana_dashboards/MongoDB_ReplSet_Summary.json (hunk 13 above)
Comment (Gu1nness) [reply src/grafana_dashboards/MongoDB_ReplSet_Summary.json:2925]: Duplicated definition of `replication_set` which is anyway removed as it doesn't make sense to keep it in this dashboard

**Comment by Gu1nness** [comment]
> Looks like there are still 20 occurances of service_name in the json. Should those still be there?
TBH : Most of the occurences are for the definition + some variables that are used only when we activate the grafana juju application (for stuff like Node name).
I don't wat to remove panels, so I rather keep it that way were it's used, but it has only the "All" value which is possible.

**Review by MiaAltieri — APPROVED** [review]

**Review by Mehdi-Bendriss — APPROVED** [review]
